src/json_core.rs

```diff
@@ -1,29 +1,98 @@
 /// as serialization/deserialization payload format.
-pub struct JsonCoreSlash;
+pub trait JsonCoreSlash: Miniconf {
+    /// Update an element by path.
+    ///
+    /// # Args
+    /// * `path` - The path to the element.
+    /// * `data` - The serialized data making up the content.
+    ///
+    /// # Returns
+    /// The number of bytes consumed from `data` or an [Error].
+    fn set_json(
+        &mut self,
+        path: &str,
+        data: &[u8],
+    ) -> core::result::Result<usize, Error<de::Error>>;
 
-impl<T> SerDe<JsonCoreSlash> for T
-where
-    T: Miniconf,
-{
-    const SEPARATOR: char = '/';
-    type DeError = de::Error;
-    type SerError = ser::Error;
+    /// Retrieve a serialized value by path.
+    ///
+    /// # Args
+    /// * `path` - The path to the element.
+    /// * `data` - The buffer to serialize the data into.
+    ///
+    /// # Returns
+    /// The number of bytes used in the `data` buffer or an [Error].
+    fn get_json(
+        &self,
+        path: &str,
+        data: &mut [u8],
+    ) -> core::result::Result<usize, Error<ser::Error>>;
 
-    fn set(&mut self, path: &str, data: &[u8]) -> Result<usize, Error<Self::DeError>> {
+    /// Update an element by indices.
+    ///
+    /// # Args
+    /// * `indices` - The indices to the element.
+    /// * `data` - The serialized data making up the content.
+    ///
+    /// # Returns
+    /// The number of bytes consumed from `data` or an [Error].
+    fn set_json_by_index(
+        &mut self,
+        indices: &[usize],
+        data: &[u8],
+    ) -> core::result::Result<usize, Error<de::Error>>;
+
+    /// Retrieve a serialized value by indices.
+    ///
+    /// # Args
+    /// * `indices` - The indices to the element.
+    /// * `data` - The buffer to serialize the data into.
+    ///
+    /// # Returns
+    /// The number of bytes used in the `data` buffer or an [Error].
+    fn get_json_by_index(
+        &self,
+        indices: &[usize],
+        data: &mut [u8],
+    ) -> core::result::Result<usize, Error<ser::Error>>;
+}
+
+impl<T: Miniconf> JsonCoreSlash for T {
+    fn set_json(&mut self, path: &str, data: &[u8]) -> Result<usize, Error<de::Error>> {
+        let mut de = de::Deserializer::new(data);
+        self.set_by_key(path.split('/').skip(1), &mut de)?;
+        de.end().map_err(Error::PostDeserialization)
+    }
+
+    fn get_json(&self, path: &str, data: &mut [u8]) -> Result<usize, Error<ser::Error>> {
+        let mut ser = ser::Serializer::new(data);
+        self.get_by_key(path.split('/').skip(1), &mut ser)?;
+        Ok(ser.end())
+    }
+
+    fn set_json_by_index(
+        &mut self,
+        indices: &[usize],
+        data: &[u8],
+    ) -> Result<usize, Error<de::Error>> {
         let mut de = de::Deserializer::new(data);
-        self.set_path(&mut path.split(Self::SEPARATOR).skip(1), &mut de)?;
+        self.set_by_key(indices.iter().copied(), &mut de)?;
```

Comment:
Why do we have to copy the interator? Seems like we should restructure the function args instead because we don't use `indices` anywhere else

Reply:
This is not copying the iterator but turning `Iterator<&usize>` into `Iterator<usize>`.